From: Matthias Klose <doko@debian.org>
Date: Sat, 29 Oct 2022 21:23:32 -0400
Subject: Build reproduceable date and time into build info

Build information is encoded into getbuildinfo.o at build time.
Use the date and time from the debian changelog, to make this reproduceable.
Forwarded: no
---
 Makefile.pre.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index ac3d37c..e284e10 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -797,6 +797,8 @@ Modules/getbuildinfo.o: $(PARSER_OBJS) \
 	      -DGITVERSION="\"`LC_ALL=C $(GITVERSION)`\"" \
 	      -DGITTAG="\"`LC_ALL=C $(GITTAG)`\"" \
 	      -DGITBRANCH="\"`LC_ALL=C $(GITBRANCH)`\"" \
+	      $(if $(BUILD_DATE),-DDATE='"$(BUILD_DATE)"') \
+	      $(if $(BUILD_TIME),-DTIME='"$(BUILD_TIME)"') \
 	      -o $@ $(srcdir)/Modules/getbuildinfo.c
 
 Modules/getpath.o: $(srcdir)/Modules/getpath.c Makefile
